transport.go

```diff
@@ -0,0 +1,123 @@
+package lumigotracer
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"net/http"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/propagation"
+	semconv "go.opentelemetry.io/otel/semconv/v1.4.0"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Transport struct {
+	rt         http.RoundTripper
+	provider   trace.TracerProvider
+	propagator propagation.TextMapPropagator
+}
+
+func NewTransport(transport http.RoundTripper) *Transport {
+	return &Transport{
+		rt:         transport,
+		provider:   otel.GetTracerProvider(),
+		propagator: otel.GetTextMapPropagator(),
+	}
+}
+
+func (t *Transport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	traceCtx, span := t.provider.Tracer("lumigo").Start(req.Context(), "HttpSpan")
+
+	req = req.WithContext(traceCtx)
+	span.SetAttributes(semconv.HTTPClientAttributesFromHTTPRequest(req)...)
+	span.SetAttributes(semconv.HTTPTargetKey.String(req.URL.Path))
+	span.SetAttributes(semconv.HTTPHostKey.String(req.URL.Host))
+	t.propagator.Inject(traceCtx, propagation.HeaderCarrier(req.Header))
+
+	if req.Body != nil {
+		bodyBytes, bodyErr := io.ReadAll(req.Body)
+		if bodyErr != nil {
+			return
+		}
+		span.SetAttributes(attribute.String("http.request_body", string(bodyBytes)))
+		// restore body
+		req.Body = ioutil.NopCloser(bytes.NewBuffer(bodyBytes))
+	}
+
+	reqHeaders := make(map[string]string)
+	for k, values := range req.Header {
+		for _, value := range values {
+			reqHeaders[k] = value
+		}
+	}
+	headersJson, err := json.Marshal(reqHeaders)
+	if err != nil {
+		logger.WithError(err).Error("failed to fetch request headers")
+	}
+	span.SetAttributes(attribute.String("http.request_headers", string(headersJson)))
+
+	resp, err = t.rt.RoundTrip(req)
+	if err != nil {
+		return
```

Comment:
if there was an error in the HTTP - we will not get it?
Check out the function `add_exception_to_span` in `lumigo_tracer/spans_container.py:211`

Reply:
https://github.com/lumigo-io/python_tracer/blob/d45303488f220cd7e93eddf108f4f2b13482b07b/src/lumigo_tracer/spans_container.py#L211

Reply:
Saar it right, it is important, we should catch 4xx and 5xx responses.